### DISCUSSION TITLE
How to mix(resource & navigationgroup) the order of navigation items?

### DISCUSSION THREAD

**Discussion by rajentrivedi** [filamentphp/filament#12150] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

Hi,

I am working on a project, where some of my resources are grouped and some are not grouped, for group I need icons for that I have added below given code to my AdminPanelProvider.php file.

    `->navigationGroups([
                NavigationGroup::make()
                     ->label('Users')
                     ->icon('heroicon-o-user'),
                NavigationGroup::make()
                     ->label('Services')
                     ->icon('eos-service-instance-o'),
                NavigationGroup::make()
                     ->label('Admin')
                     ->icon('eos-admin-o'),
                NavigationGroup::make()
                     ->label('Master')
                     ->icon('eos-master'),
                NavigationGroup::make()
                     ->label('Content Management')
                     ->icon('heroicon-o-globe-alt'),
                ])`
            
            the look of my navigation items are shown in below image
            
![Screenshot from 2024-04-02 21-25-10](https://github.com/filamentphp/filament/assets/69707769/679c8d06-9bb8-4dbb-975c-b090e09ae670)

All the grouped resources are showing below not grouped resources, I need to customize the navigation item orders, I want some grouped resource on top then ungrouped resouce item (mix).

How can I achieve this?

Thank you.

**Comment by rajentrivedi** [comment, marked as answer]
Hi,

I have done it,


    `$panel->navigation(function (NavigationBuilder $builder): NavigationBuilder {
                return $builder->groups([
                    NavigationGroup::make('Users')
                     ->label('Users')
                     ->icon('heroicon-o-user')
                     ->items([
                         ...UserResource::getNavigationItems(),
                         ...CustomerResource::getNavigationItems(),
                         ...GardenerResource::getNavigationItems()
                     ]),
                     NavigationGroup::make('Services')
                     ->label('Services')
                     ->icon('eos-service-instance-o')
                     ->items([
                        ...CategoryResource::getNavigationItems(),
                        ...ProductResource::getNavigationItems()
                     ]),

                    NavigationGroup::make()
                        ->collapsible(false)
                        ->items([
                            ...RequestResource::getNavigationItems(),
                            // Message Resource
                            ...EmailTamplateResource::getNavigationItems(),
                        ]),
                    NavigationGroup::make('Master')
                     ->label('Master')
                     ->icon('eos-master')
                     ->items([
                        ...GardenSizeResource::getNavigationItems(),
                        ...GardenTypeResource::getNavigationItems()
                     ]),
                     NavigationGroup::make()
                        ->collapsible(false)
                        ->items([
                            // User Referal Resource
                            ...SubscriptionResouceResource::getNavigationItems(),
                        ]),
                    NavigationGroup::make('Admin')
                     ->label('Admin')
                     ->icon('eos-admin-o')
                     ->items([
                        ...AdminResource::getNavigationItems(),
                        // ...GardenTypeResource::getNavigationItems()
                     ]),
                     NavigationGroup::make('Content Management')
                     ->label('Content Management')
                     ->icon('heroicon-o-globe-alt')
                     ->items([
                        ...SlideResouceResource::getNavigationItems(),
                        ...CmsPageResource::getNavigationItems(),
                     ]),
                ]);
            });`

**Reply by nlvpatrick** [reply]
is this working?
